File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/CustomWebChromeClient.java
package jp.hazuki.yuzubrowser.webkit;

import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Message;
import android.view.View;
import android.webkit.ConsoleMessage;
import android.webkit.GeolocationPermissions;
import android.webkit.JsPromptResult;
import android.webkit.JsResult;
import android.webkit.ValueCallback;
import android.webkit.WebChromeClient;
import android.webkit.WebView;

public class CustomWebChromeClient extends WebChromeClient {
    public Bitmap getDefaultVideoPoster() {
        return null;
    }

    public View getVideoLoadingProgressView() {
        return null;
    }

    public void getVisitedHistory(ValueCallback<String[]> callback) {
    }

    public void onCloseWindow(CustomWebView window) {
    }

    @Override
    public void onCloseWindow(WebView window) {
        if (window instanceof CustomWebView)
            onCloseWindow((CustomWebView) window);
    }

    public boolean onConsoleMessage(ConsoleMessage consoleMessage) {
        return false;
    }

    public boolean onCreateWindow(CustomWebView view, boolean isDialog, boolean isUserGesture, Message resultMsg) {
        return false;
    }

    @Override
    public boolean onCreateWindow(WebView view, boolean isDialog, boolean isUserGesture, Message resultMsg) {
        return view instanceof CustomWebView && onCreateWindow((CustomWebView) view, isDialog, isUserGesture, resultMsg);
    }

    public void onGeolocationPermissionsHidePrompt() {
    }

    public void onGeolocationPermissionsShowPrompt(String origin, GeolocationPermissions.Callback callback) {
    }

    public void onHideCustomView() {
    }

    public boolean onJsAlert(CustomWebView view, String url, String message, JsResult result) {
        return false;
    }

    @Override
    public boolean onJsAlert(WebView view, String url, String message, JsResult result) {
        return view instanceof CustomWebView && onJsAlert((CustomWebView) view, url, message, result);
    }

    public boolean onJsBeforeUnload(CustomWebView view, String url, String message, JsResult result) {
        return false;
    }

    @Override
    public boolean onJsBeforeUnload(WebView view, String url, String message, JsResult result) {
        return view instanceof CustomWebView && onJsBeforeUnload((CustomWebView) view, url, message, result);
    }

    public boolean onJsConfirm(CustomWebView view, String url, String message, JsResult result) {
        return false;
    }

    @Override
    public boolean onJsConfirm(WebView view, String url, String message, JsResult result) {
        return view instanceof CustomWebView && onJsConfirm((CustomWebView) view, url, message, result);
    }

    public boolean onJsPrompt(CustomWebView view, String url, String message, String defaultValue, JsPromptResult result) {
        return false;
    }

    @Override
    public boolean onJsPrompt(WebView view, String url, String message, String defaultValue, JsPromptResult result) {
        return view instanceof CustomWebView && onJsPrompt((CustomWebView) view, url, message, defaultValue, result);
    }

    public void onProgressChanged(CustomWebView view, int newProgress) {
    }

    @Override
    public void onProgressChanged(WebView view, int newProgress) {
        if (view instanceof CustomWebView)
            onProgressChanged((CustomWebView) view, newProgress);
    }

    public void onReceivedIcon(CustomWebView view, Bitmap icon) {
    }

    @Override
    public void onReceivedIcon(WebView view, Bitmap icon) {
        if (view instanceof CustomWebView)
            onReceivedIcon((CustomWebView) view, icon);
    }

    public void onReceivedTitle(CustomWebView view, String title) {
    }

    @Override
    public void onReceivedTitle(WebView view, String title) {
        if (view instanceof CustomWebView)
            onReceivedTitle((CustomWebView) view, title);
    }

    public void onReceivedTouchIconUrl(CustomWebView view, String url, boolean precomposed) {
    }

    @Override
    public void onReceivedTouchIconUrl(WebView view, String url, boolean precomposed) {
        if (view instanceof CustomWebView)
            onReceivedTouchIconUrl((CustomWebView) view, url, precomposed);
    }

    public void onRequestFocus(CustomWebView view) {
    }

    @Override
    public void onRequestFocus(WebView view) {
        if (view instanceof CustomWebView)
            onRequestFocus((CustomWebView) view);
    }

    public void onShowCustomView(View view, CustomViewCallback callback) {
    }

    public void openFileChooser(ValueCallback<Uri> uploadMsg, String acceptType, String capture) {
        uploadMsg.onReceiveValue(null);
    }

    public void openFileChooser(ValueCallback<Uri> uploadMsg, final String acceptType) {
        uploadMsg.onReceiveValue(null);
    }

    public void openFileChooser(ValueCallback<Uri> uploadMsg) {
        uploadMsg.onReceiveValue(null);
    }
}

File: app/src/main/java/jp/hazuki/yuzubrowser/webkit/CustomWebChromeClient.kt
package jp.hazuki.yuzubrowser.webkit

import android.graphics.Bitmap
import android.os.Message
import android.view.View
import android.webkit.*

open class CustomWebChromeClient : WebChromeClient() {
    override fun getDefaultVideoPoster(): Bitmap? = null

    override fun getVideoLoadingProgressView(): View? = null

    override fun getVisitedHistory(callback: ValueCallback<Array<String>>) {}

    open fun onCloseWindow(web: CustomWebView) {}

    override fun onCloseWindow(window: WebView) {
        if (window is CustomWebView)
            onCloseWindow(window as CustomWebView)
    }

    override fun onConsoleMessage(consoleMessage: ConsoleMessage): Boolean = false

    open fun onCreateWindow(view: CustomWebView, isDialog: Boolean, isUserGesture: Boolean, resultMsg: Message): Boolean = false

    override fun onCreateWindow(view: WebView, isDialog: Boolean, isUserGesture: Boolean, resultMsg: Message): Boolean =
            view is CustomWebView && onCreateWindow(view as CustomWebView, isDialog, isUserGesture, resultMsg)

    override fun onGeolocationPermissionsHidePrompt() {}

    override fun onGeolocationPermissionsShowPrompt(origin: String, callback: GeolocationPermissions.Callback) {}

    override fun onHideCustomView() {}

    open fun onJsAlert(view: CustomWebView, url: String, message: String, result: JsResult): Boolean = false

    override fun onJsAlert(view: WebView, url: String, message: String, result: JsResult): Boolean =
            view is CustomWebView && onJsAlert(view as CustomWebView, url, message, result)

    open fun onJsBeforeUnload(view: CustomWebView, url: String, message: String, result: JsResult): Boolean = false

    override fun onJsBeforeUnload(view: WebView, url: String, message: String, result: JsResult): Boolean =
            view is CustomWebView && onJsBeforeUnload(view as CustomWebView, url, message, result)

    open fun onJsConfirm(view: CustomWebView, url: String, message: String, result: JsResult): Boolean = false

    override fun onJsConfirm(view: WebView, url: String, message: String, result: JsResult): Boolean =
            view is CustomWebView && onJsConfirm(view as CustomWebView, url, message, result)

    open fun onJsPrompt(view: CustomWebView, url: String, message: String, defaultValue: String, result: JsPromptResult): Boolean = false

    override fun onJsPrompt(view: WebView, url: String, message: String, defaultValue: String, result: JsPromptResult): Boolean =
            view is CustomWebView && onJsPrompt(view as CustomWebView, url, message, defaultValue, result)

    open fun onProgressChanged(web: CustomWebView, newProgress: Int) {}

    override fun onProgressChanged(view: WebView, newProgress: Int) {
        if (view is CustomWebView)
            onProgressChanged(view as CustomWebView, newProgress)
    }

    open fun onReceivedIcon(web: CustomWebView, icon: Bitmap) {}

    override fun onReceivedIcon(view: WebView, icon: Bitmap) {
        if (view is CustomWebView)
            onReceivedIcon(view as CustomWebView, icon)
    }

    open fun onReceivedTitle(web: CustomWebView, title: String) {}

    override fun onReceivedTitle(view: WebView, title: String) {
        if (view is CustomWebView)
            onReceivedTitle(view as CustomWebView, title)
    }

    open fun onReceivedTouchIconUrl(view: CustomWebView, url: String, precomposed: Boolean) {}

    override fun onReceivedTouchIconUrl(view: WebView, url: String, precomposed: Boolean) {
        if (view is CustomWebView)
            onReceivedTouchIconUrl(view as CustomWebView, url, precomposed)
    }

    open fun onRequestFocus(web: CustomWebView) {}

    override fun onRequestFocus(view: WebView) {
        if (view is CustomWebView)
            onRequestFocus(view as CustomWebView)
    }

    override fun onShowCustomView(view: View, callback: WebChromeClient.CustomViewCallback) {}
}
